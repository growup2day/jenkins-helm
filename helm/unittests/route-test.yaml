templates:
  - route.yaml
tests:
  - it: manifest should match snapshot
    set:
      domain: dummy.domain
    asserts:
      - matchSnapshot: {}
  - it: should include specified route annotations
    set:
      routeAnnotations:
        key1: value1
        key2: value2
    asserts:
      - matchSnapshot:
          path: metadata.annotations
  - it: should allow setting a completely custom route host
    set:
      domain: dummy.domain
      route:
        host: my.custom.domain
    asserts:
      - equal:
          path: spec.host
          value: my.custom.domain
  - it: should allow route host to be set without domain specified
    set:
      route:
        host: my.custom.domain
    asserts:
      - equal:
          path: spec.host
          value: my.custom.domain
  - it: should compute a route host using domain and subdomain
    set:
      domain: dummy.domain
      route:
        subDomain: subdomain
    asserts:
      - equal:
          path: spec.host
          value: subdomain.dummy.domain
  - it: should generate a full route host when not explicitly specified
    set:
      domain: dummy.domain
    asserts:
      - equal:
          path: spec.host
          value: RELEASE-NAME-jenkins-NAMESPACE.dummy.domain
  - it: full custom route should take precidence over subdomain route when both are specified
    set:
      domain: dummy.domain
      route:
        host: my.custom.domain
        subDomain: subdomain
    asserts:
      - equal: 
          path: spec.host
          value: my.custom.domain