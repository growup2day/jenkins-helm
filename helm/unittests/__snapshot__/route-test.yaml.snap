manifest should match snapshot:
  1: |
    apiVersion: route.openshift.io/v1
    kind: Route
    metadata:
      labels:
        app: RELEASE-NAME-jenkins
        app.kubernetes.io/managed-by: Helm
        helm.sh/chart: jenkins-0.7.0
        name: RELEASE-NAME-jenkins
        release-name: RELEASE-NAME
      name: RELEASE-NAME-jenkins
    spec:
      host: RELEASE-NAME-jenkins-NAMESPACE.dummy.domain
      tls:
        insecureEdgeTerminationPolicy: Redirect
        termination: edge
      to:
        kind: Service
        name: RELEASE-NAME-jenkins
should include specified route annotations:
  1: |
    key1: value1
    key2: value2
